### ISSUE TITLE
Would love to try your tool

### ISSUE THREAD

**Issue by Devin-Licastro** [GeneralVimes/AS32JS#1]
Hey there, would love to give your tool a shot and see what comes out! Looks very interesting.

I've tried first putting my src directory into a new source_as3 folder like this:
![image](https://github.com/GeneralVimes/AS32JS/assets/40781839/d682bc48-7f0c-4832-8581-d432848485ec)

Then when running python class_processor.py I get:

![image](https://github.com/GeneralVimes/AS32JS/assets/40781839/b7462050-bec3-4f7f-8bb9-ca891b8a005b)

Any idea what I should be doing differently?

I think my use case may be a bit too tricky for auto conversion, (it uses local sqlite database and lots of api calls and stuff) but it would be great to see how far I can get with it. It would be amazing to have an html 5 version of my app even if it took some work.

Thanks for any tips

**Comment by GeneralVimes** [comment]
Thank you for trying! Are there only .as3 files in the source_as3 folder?

**Comment by GeneralVimes** [comment]
Also, could you please send me to general@cardswars.com the file ExtendedTimer.as?

**Comment by Devin-Licastro** [comment]
Hey! Really sorry I didn't get back to you on this, I had notifications off and then got distracted by other things and forgot to follow up.

So I'm still eager to try, if you're happy to help :)

I've removed the ColorFilter and ExtendedTimer classes as it turns out my project wasn't using them anymore. 
The next file it tries to process is called Helper.as it gets stuck there. That file is full of a bunch of random functions for processing dates and images and stuff. 

![image](https://github.com/GeneralVimes/AS32JS/assets/40781839/12a687a0-8df3-4d76-90b9-68f7e87e1032)

If I temporarily remove that file then it makes it much further, but still seems to be getting a lot of errors:
![image](https://github.com/GeneralVimes/AS32JS/assets/40781839/a1e86c13-59c5-4877-b0c8-a76451ca9e8d)
![image](https://github.com/GeneralVimes/AS32JS/assets/40781839/372498e1-6e00-4493-b565-461588c4ddba)

Is there some process I can follow to try to help it find parents when it is struggling?

I've attached both the Helper and AnswerData_WA files here 
[both_files.zip](https://github.com/user-attachments/files/16125530/both_files.zip)
if you'd like to take a look.

Thanks,

**Comment by GeneralVimes** [comment]
Oh, cool! Thank you, I'll take a look at these files and will see, what should I add into my python script. In the meanwhile I also improved the conversion algorithm from my own cases, going to update the tool soon

**Comment by GeneralVimes** [comment]
@Devin-Licastro I checked how my script processes your files. I think the errors are because there are functions inside of functions. My script was not developed to handle these cases as I don't use such constructions in my games

**Comment by Devin-Licastro** [comment]
Hmm okay, yeah unfortunately I have that in quite a few places in my project. So the only way forward would be the restructure all of these ya?

**Comment by GeneralVimes** [comment]
Yes, because now the tool looks through the classes, makes a list of methods, and then replaces their signatures and adds this. before their calls. So, having function inside a function breaks this pattern.
Alternatively, you might try to auto-convert your files, and then manually make the changes
